src/cdk/table/table.ts

```diff
@@ -493,19 +551,24 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
    * within the data row view container.
    */
   private _insertRow(rowData: T, index: number) {
-    const row = this._getRowDef(rowData, index);
-
-    // Row context that will be provided to both the created embedded row view and its cells.
+    const rowDef = this._getRowDef(rowData, index);
     const context: CdkCellOutletRowContext<T> = {$implicit: rowData};
+    this._renderRow(this._rowPlaceholder, rowDef, context, index);
+  }
 
-    // TODO(andrewseguin): add some code to enforce that exactly one
-    //   CdkCellOutlet was instantiated as a result  of `createEmbeddedView`.
-    this._rowPlaceholder.viewContainer.createEmbeddedView(row.template, context, index);
+  /**
+   * Creates a new row template in the outlet and fills it with the set of cell templates.
+   * Optionally takes a context to provide to the row and cells, as well as an optional index
+   * of where to place the new row template in the outlet.
+   */
+  private _renderRow(
+      outlet: any, rowDef: BaseRowDef, context: CdkCellOutletRowContext<T> = {}, index = 0) {
```

Comment:
`outlet` should be able to be more specific than `any`

Reply:
Done - added a base class for the row outlets